### PR TITLE
Create jupytercad-* packages

### LINKED ISSUE #218: Split the package into multiple sub-packages
In addition to https://github.com/QuantStack/jupytercad/issues/215, we should make jupytercad more modular, with the main `jupytercad` package ending up being a meta-package that simply installs multiple sub-packages.

- `jupytercad-lab`: would be the lab extension for jupytercad
- `jupytercad-app`: would be the standalone cad application
- `jupytercad`: would be the meta-package that depends on `jupytercad-lab` and `jupytercad-app`
- Packages for each file type

### PR DESCRIPTION
**Pull request by trungleduc**
## Reference
- Closes #218 

## Code changes
- New python packages:
  - `jupytercad-core`: package contains core plugins for worker registry, jcad file support, doc tracker and annotation
  - `jupytercad-lab`: package contains plugin for lab interface and notebook support
  - `jupytercad-app`: the standalone Jupytercad application
- New js packages:
  - `@jupytercad/schema`: package contains the schema of jcad file and related models/interfaces
  - `@jupytercad/opencascade`: the custom opencascade.js build for Jupytercad
  - `@jupytercad/occ-worker`: the opencascade worker of Jupytercad.
  - `@jupytercad/base`: package contains the UI components.

## Remaining tasks:

- [ ] Create meta package jupytercad
- [ ] Update CI 
- [ ] Update jupyter-releaser config
- [ ] Create package jupytercad-fcstd

### CODE REVIEW

File: python/jupytercad-app/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+[build-system]
+requires = [
+    "hatchling>=1.5.0",
+    "jupyterlab>=4.0.0,<5",
+    "hatch-nodejs-version>=0.3.2",
+]
+build-backend = "hatchling.build"
+
+[project]
+dependencies = ["jupyter_server>=2.0.1,<3", "jupytercad_core>=0.3.3,<0.4"]
```
Comment (martinRenou) [inline python/jupytercad-app/pyproject.toml:10]: Are we missing `jupytercad_lab` here? I suppose the `jupytercad_app` package without the `jupytercad_lab` lab plugins would not work?

File: jupytercad/__init__.py
Comment (martinRenou) [inline jupytercad/__init__.py:1]: There are still some Python modules in there like `fcstd_ydoc.py`, should we move those in `jupytercad-fcstd`? Do you want to do that in a separate PR?

**Review by martinRenou — COMMENTED** [review]
Comment on the Python packages names. I would change the distribution names to `jupytercad-core` `jupytercad-app` etc (dash instead of underscore)

File: jupytercad/__init__.py
Comment (trungleduc) [reply jupytercad/__init__.py:1]: yep, I will move those to `jupytercad-fcstd` repo in the next PR

**Review by martinRenou — APPROVED** [review]
LGTM thanks! Let's merge and fix the CI separately
